Python/suggestions.c

```diff
@@ -90,38 +197,39 @@ calculate_suggestions(PyObject *dir,
     if (name_str == NULL) {
         return NULL;
     }
+    size_t *work_buffer = PyMem_Calloc(name_size, sizeof(size_t));
```

Comment:
I think that as we work with max length strings we can statically allocate a fixed array here, which simplifies the code a lot